flink-runtime/src/main/java/org/apache/flink/runtime/operators/BatchTask.java

```diff
@@ -1150,8 +1149,8 @@ else if (index < 0 || index >= this.driver.getNumberOfDriverComparators()) {
 	 * @return The string for logging.
 	 */
 	public static String constructLogString(String message, String taskName, AbstractInvokable parent) {
-		return message + ":  " + taskName + " (" + (parent.getEnvironment().getIndexInSubtaskGroup() + 1) +
-				'/' + parent.getEnvironment().getNumberOfSubtasks() + ')';
+		return message + ":  " + taskName + " (" + (parent.getEnvironment().getTaskInfo().getIndexOfThisSubtask() + 1) +
```

Comment:
Can replace the subtask concatenation with `getTaskInfo().getTaskNameWithSubtasks()`.